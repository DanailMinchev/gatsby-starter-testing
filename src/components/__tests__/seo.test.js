import React from "react"
import { render, waitFor } from "@testing-library/react"
import { useStaticQuery } from "gatsby" // mocked

import SEO, { PureSEO } from "../seo"

describe(`SEO`, () => {
  beforeEach(() => {
    useStaticQuery.mockImplementationOnce(() => ({
      site: {
        siteMetadata: {
          title: "Gatsby Testing Starter",
          description: "A simple starter with configured testing",
          author: "@DanailMinchev",
        },
      },
    }))
  })

  it(`renders title`, async () => {
    const title = `Hello World`

    render(<SEO title={title} />)

    await waitFor(() =>
      expect(document.title).toEqual(`${title} | Gatsby Testing Starter`)
    )
  })

  it(`renders lang attribute`, async () => {
    const title = `Hello World`
    const lang = `ru`

    render(<SEO title={title} lang={lang} />)

    await waitFor(() =>
      expect(document.querySelector(`html`)).toHaveAttribute(`lang`, lang)
    )
  })

  it(`renders meta tags`, async () => {
    const description = `Description here ...`
    const lang = `bg`
    const meta = [
      {
        name: `name_here`,
        content: `content here ...`,
      },
      {
        property: `property:here`,
        content: `content here ...`,
      },
    ]
    const title = `Title here ...`

    render(
      <SEO description={description} lang={lang} meta={meta} title={title} />
    )

    await waitFor(() => {
      expect(document.querySelectorAll(`meta`)).toHaveLength(10)
    })
    expect(document.querySelector("head")).toMatchSnapshot()
  })

  it(`renders meta tags with defaults`, async () => {
    const title = `Hello World`

    render(<SEO title={title} />)

    await waitFor(() => {
      expect(document.querySelectorAll(`meta`)).toHaveLength(8)
    })
    expect(document.querySelector("head")).toMatchSnapshot()
  })
})

describe(`PureSEO`, () => {
  const data = {
    site: {
      siteMetadata: {
        title: "Gatsby Testing Starter",
        description: "A simple starter with configured testing",
        author: "@DanailMinchev",
      },
    },
  }

  it(`renders title`, async () => {
    const title = `Hello World`

    render(<PureSEO data={data} title={title} />)

    await waitFor(() =>
      expect(document.title).toEqual(`${title} | Gatsby Testing Starter`)
    )
  })

  it(`renders lang attribute`, async () => {
    const title = `Hello World`
    const lang = `ru`

    render(<PureSEO data={data} title={title} lang={lang} />)

    await waitFor(() =>
      expect(document.querySelector(`html`)).toHaveAttribute(`lang`, lang)
    )
  })

  it(`renders meta tags`, async () => {
    const description = `Description here ...`
    const lang = `bg`
    const meta = [
      {
        name: `name_here`,
        content: `content here ...`,
      },
      {
        property: `property:here`,
        content: `content here ...`,
      },
    ]
    const title = `Title here ...`

    render(
      <PureSEO
        data={data}
        description={description}
        lang={lang}
        meta={meta}
        title={title}
      />
    )

    await waitFor(() => {
      expect(document.querySelectorAll(`meta`)).toHaveLength(10)
    })
    expect(document.querySelector("head")).toMatchSnapshot()
  })

  it(`renders meta tags with defaults`, async () => {
    const title = `Hello World`

    render(<PureSEO data={data} title={title} />)

    await waitFor(() => {
      expect(document.querySelectorAll(`meta`)).toHaveLength(8)
    })
    expect(document.querySelector("head")).toMatchSnapshot()
  })
})
