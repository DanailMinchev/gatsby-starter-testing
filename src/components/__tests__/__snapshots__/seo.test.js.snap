// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PureSEO renders meta tags 1`] = `
<head>
  <title>
    Title here ... | Gatsby Testing Starter
  </title>
  <meta
    content="Description here ..."
    data-react-helmet="true"
    name="description"
  />
  <meta
    content="Title here ..."
    data-react-helmet="true"
    property="og:title"
  />
  <meta
    content="Description here ..."
    data-react-helmet="true"
    property="og:description"
  />
  <meta
    content="website"
    data-react-helmet="true"
    property="og:type"
  />
  <meta
    content="summary"
    data-react-helmet="true"
    name="twitter:card"
  />
  <meta
    content="@DanailMinchev"
    data-react-helmet="true"
    name="twitter:creator"
  />
  <meta
    content="Title here ..."
    data-react-helmet="true"
    name="twitter:title"
  />
  <meta
    content="Description here ..."
    data-react-helmet="true"
    name="twitter:description"
  />
  <meta
    content="content here ..."
    data-react-helmet="true"
    name="name_here"
  />
  <meta
    content="content here ..."
    data-react-helmet="true"
    property="property:here"
  />
</head>
`;

exports[`PureSEO renders meta tags with defaults 1`] = `
<head>
  <title>
    Hello World | Gatsby Testing Starter
  </title>
  <meta
    content="A simple starter with configured testing"
    data-react-helmet="true"
    name="description"
  />
  <meta
    content="Hello World"
    data-react-helmet="true"
    property="og:title"
  />
  <meta
    content="A simple starter with configured testing"
    data-react-helmet="true"
    property="og:description"
  />
  <meta
    content="website"
    data-react-helmet="true"
    property="og:type"
  />
  <meta
    content="summary"
    data-react-helmet="true"
    name="twitter:card"
  />
  <meta
    content="@DanailMinchev"
    data-react-helmet="true"
    name="twitter:creator"
  />
  <meta
    content="Hello World"
    data-react-helmet="true"
    name="twitter:title"
  />
  <meta
    content="A simple starter with configured testing"
    data-react-helmet="true"
    name="twitter:description"
  />
</head>
`;

exports[`SEO renders meta tags 1`] = `
<head>
  <title>
    Title here ... | Gatsby Testing Starter
  </title>
  <meta
    content="Description here ..."
    data-react-helmet="true"
    name="description"
  />
  <meta
    content="Title here ..."
    data-react-helmet="true"
    property="og:title"
  />
  <meta
    content="Description here ..."
    data-react-helmet="true"
    property="og:description"
  />
  <meta
    content="website"
    data-react-helmet="true"
    property="og:type"
  />
  <meta
    content="summary"
    data-react-helmet="true"
    name="twitter:card"
  />
  <meta
    content="@DanailMinchev"
    data-react-helmet="true"
    name="twitter:creator"
  />
  <meta
    content="Title here ..."
    data-react-helmet="true"
    name="twitter:title"
  />
  <meta
    content="Description here ..."
    data-react-helmet="true"
    name="twitter:description"
  />
  <meta
    content="content here ..."
    data-react-helmet="true"
    name="name_here"
  />
  <meta
    content="content here ..."
    data-react-helmet="true"
    property="property:here"
  />
</head>
`;

exports[`SEO renders meta tags with defaults 1`] = `
<head>
  <title>
    Hello World | Gatsby Testing Starter
  </title>
  <meta
    content="A simple starter with configured testing"
    data-react-helmet="true"
    name="description"
  />
  <meta
    content="Hello World"
    data-react-helmet="true"
    property="og:title"
  />
  <meta
    content="A simple starter with configured testing"
    data-react-helmet="true"
    property="og:description"
  />
  <meta
    content="website"
    data-react-helmet="true"
    property="og:type"
  />
  <meta
    content="summary"
    data-react-helmet="true"
    name="twitter:card"
  />
  <meta
    content="@DanailMinchev"
    data-react-helmet="true"
    name="twitter:creator"
  />
  <meta
    content="Hello World"
    data-react-helmet="true"
    name="twitter:title"
  />
  <meta
    content="A simple starter with configured testing"
    data-react-helmet="true"
    name="twitter:description"
  />
</head>
`;
